lib/chai/core/assertions.js

```diff
@@ -1634,32 +1634,38 @@ module.exports = function (chai, _) {
    * @name change
    * @alias changes
    * @alias Change
-   * @param {String} object
+   * @param {String} target
    * @param {String} property name _optional_
    * @param {String} message _optional_
    * @namespace BDD
    * @api public
    */
 
-  function assertChanges (object, prop, msg) {
+  function assertChanges (target, prop, msg) {
     if (msg) flag(this, 'message', msg);
     var fn = flag(this, 'object');
     new Assertion(fn).is.a('function');
 
     var initial;
     if (!prop) {
-      new Assertion(object).is.a('function');
-      initial = object();
+      new Assertion(target).is.a('function');
+      initial = target();
     } else {
-      new Assertion(object, msg).to.have.property(prop);
-      initial = object[prop];
+      new Assertion(target, msg).to.have.property(prop);
+      initial = target[prop];
     }
 
     fn();
 
-    var final = prop === undefined ? object() : object[prop];
+    var final = prop === undefined ? target() : target[prop];
     var msgObj = prop === undefined ? initial : '.' + prop;
 
+    // This gets flagged because of the .by(delta) assertion
+    flag(this, 'msgObj', msgObj);
+    flag(this, 'initial', initial);
+    flag(this, 'final', final);
```

Comment:
I'm not sold on the names of these flags. They're very likely only going to be used by the increase/decrease/change set - having then more specific to that would be good.